src/h5web/visualizations/heatmap/HeatmapProvider.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, ReactElement, ReactNode } from 'react';
+import { Dims, AxisOffsets } from './models';
+
+export interface HeatmapProps {
+  dims: Dims;
+  data: number[][];
+  axisOffsets: AxisOffsets;
+}
+
+export const HeatmapContext = createContext<HeatmapProps | undefined>(
+  undefined
+);
+
+type Props = {
+  dims: Dims;
+  data: number[][];
+  children: ReactNode;
+};
+
+function HeatmapProvider(props: Props): ReactElement {
+  const { dims, data, children } = props;
+
+  return (
+    <HeatmapContext.Provider value={{ dims, data, axisOffsets: [72, 36] }}>
+      {children}
+    </HeatmapContext.Provider>
+  );
+}
+
+export default HeatmapProvider;
```

Comment:
This is the context provider. It just forwards the props, along with`axisOffsets` for convenience.

Reply:
hmm smart 💯